src/openssl/ssl/context.cr

```diff
@@ -1,22 +1,40 @@
 class OpenSSL::SSL::Context
-  def self.default : self
-    @@default ||= new
-  end
-
   def initialize
     @handle = LibSSL.ssl_ctx_new(LibSSL.sslv23_method)
+    self.verify_mode = OpenSSL::SSL::VerifyMode::PEER
```

Comment:
Maybe we should only set the verify mode from HTTP::Client (and other protocols)?